cpp/ql/test/library-tests/dataflow/DefaultTaintTracking/stl.cpp

```diff
@@ -16,9 +16,15 @@ namespace std
 	
 	template<class charT, class traits = char_traits<charT>, class Allocator = allocator<charT> >
 	class basic_string {
+	private:
+		const char* p;
+
 	public:
 		explicit basic_string(const Allocator& a = Allocator());
-		basic_string(const charT* s, const Allocator& a = Allocator());
+		basic_string(const charT* s, const Allocator& a = Allocator()) :
+			p(s) 
```

Comment:
There's no way `basic_string` is implemented like this. If the data-flow modelling only works with this mock implementation, then the data-flow modelling needs to be fixed. We've committed to supporting `basic_string`, so this very important constructor needs to work.

Reply:
Oops, another leftover experiment. I've fixed this for real by fixing taint flow to qualifier objects in `DefaultTaintTracking`.